### PR TITLE
yandex-browser:  23.7.1.1215-1 -> 23.9.1.962-1

### PR DESCRIPTION
**Pull request by ionutnechita**
Description:
     - yandex-browser{stable,beta,corporate} is updated to:
    
        corporate = "23.7.4.985-1"
        beta = "23.9.1.1028-1"
        stable = "23.9.1.962-1"

Signed-off-by: Ionut Nechita [ionut_n2001@yahoo.com](mailto:ionut_n2001@yahoo.com)

## Description of changes

yandex-browser: 23.7.1.1215-1 -> 23.9.1.962-1

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ionutnechita** [comment]
Hello
But I should separate into several commits. But I will do it in the next few days.

That the first version will be with several packages.

**Comment by chayleaf** [comment]
Again, please don't duplicate code. Add a derivation argument:

```nix
...
, libqt5pas
, yandex-ffmpeg-codecs
, edition ? "stable"
}:

let
  version = {
    corporate = "23.5.1.804-1";
    beta = "23.7.1.1148-1";
    stable = "23.7.1.1215-1";
  }.${edition};

  sha256 = {
    stable = "sha256-AKqdwrCs9qLyy23Yqh0eN8tgxpw6S4V5plCT7vE2gic=";
    beta = "sha256-SJbuT2MnsXcqOSk4xCUokseDotjbWgAnvwnfNPF9zi4=";
    corporate = "sha256-43K8dA+TjQsvXvA2m51O5bhVPATOmSg1/GzY633npls=";
  }.${edition};

in stdenv.mkDerivation {
  pname = if edition == "stable" then "yandex-browser" else "yandex-browser-${edition}";
  inherit version;

  src = fetchurl {
    url = "http://repo.yandex.ru/yandex-browser/deb/pool/main/y/yandex-browser-${edition}/yandex-browser-${edition}_${version}_amd64.deb";
    inherit sha256;
  };

  ...

  installPhase = ''
    substituteInPlace $out/share/applications/yandex-browser.desktop --replace /usr/ $out/
    ln -sf ${yandex-ffmpeg-codecs}/lib/libffmpeg.so $out/opt/yandex/browser/libffmpeg.so
    ln -sf $out/opt/yandex/browser/yandex-browser $out/bin/yandex-browser
    ln -sf $out/opt/yandex/browser/yandex-browser $out/bin/yandex-browser-${edition}
  '';
};
```

```nix
{
  yandex-browser-beta = yandex-browser.override { edition = "beta"; };
  yandex-browser-corporate = yandex-browser.override { edition = "corporate"; };
}
```

**Review by dan4ik605743 — APPROVED** [review]

**Comment by ionutnechita** [comment]
Ready to review.
Latest updates.

File: pkgs/applications/networking/browsers/yandex-browser/default.nix [hunk 1]
```diff
@@ -108,6 +131,8 @@ stdenv.mkDerivation rec {
     pango
     stdenv.cc.cc.lib
     libqt5pas
+    qt6.qtbase
+    qt6.wrapQtAppsHook
```
Comment (wegank) [inline pkgs/applications/networking/browsers/yandex-browser/default.nix:135]: This should go to `nativeBuildInputs`.

File: pkgs/applications/networking/browsers/yandex-ffmpeg-codecs/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+stdenv.mkDerivation rec {
+  pname = "yandex-ffmpeg-codecs";
+  version = "111306";
+
+  src = fetchurl {
+    url = "https://api.snapcraft.io/api/v1/snaps/download/XXzVIXswXKHqlUATPqGCj2w2l7BxosS8_34.snap";
+    sha256 = "sha256-Dna9yFgP7JeQLAeZWvSZ+eSMX2yQbX2/+mX0QC22lYY=";
+  };
+
+  buildInputs = [ squashfsTools ];
```
Comment (wegank) [inline pkgs/applications/networking/browsers/yandex-ffmpeg-codecs/default.nix:12]:
```suggestion
  nativeBuildInputs = [ squashfsTools ];
```

File: pkgs/applications/networking/browsers/yandex-browser/default.nix [hunk 3]
```diff
@@ -47,15 +47,38 @@
...
-  pname = "yandex-browser";
-  version = "23.7.1.1148-1";
+let
+  version = {
+    corporate = "23.7.4.985-1";
+    beta = "23.9.1.1028-1";
+    stable = "23.9.1.962-1";
+  }.${edition};
+
+  sha256 = {
```
Comment (wegank) [inline pkgs/applications/networking/browsers/yandex-browser/default.nix:62]:
```suggestion
  hash = {
```

File: pkgs/applications/networking/browsers/yandex-browser/default.nix [hunk 4]
```diff
@@ -47,15 +47,38 @@
...
+in stdenv.mkDerivation rec {
+  pname = if edition == "stable" then "yandex-browser-stable" else "yandex-browser-${edition}";
+  inherit version;
+  inherit app;
 
   src = fetchurl {
-    url = "http://repo.yandex.ru/yandex-browser/deb/pool/main/y/${pname}-beta/${pname}-beta_${version}_amd64.deb";
-    sha256 = "sha256-SJbuT2MnsXcqOSk4xCUokseDotjbWgAnvwnfNPF9zi4=";
+    url = "http://repo.yandex.ru/yandex-browser/deb/pool/main/y/${pname}/${pname}_${version}_amd64.deb";
+    inherit sha256;
```
Comment (wegank) [inline pkgs/applications/networking/browsers/yandex-browser/default.nix:81]:
```suggestion
    inherit hash;
```

File: pkgs/applications/networking/browsers/yandex-browser/default.nix [hunk 5]
```diff
@@ -47,19 +47,43 @@
...
+  app = {
+    corporate = "";
+    beta = "-beta";
+    stable = "";
+  }.${edition};
+
+in stdenv.mkDerivation rec {
+  pname = if edition == "stable" then "yandex-browser-stable" else "yandex-browser-${edition}";
+  inherit version;
+  inherit app;
```
Comment (chayleaf) [inline pkgs/applications/networking/browsers/yandex-browser/default.nix:77]: why is `inherit app` here? (also, it can be merged with `inherit version`)

Comment (wegank) [reply pkgs/applications/networking/browsers/yandex-browser/default.nix]: Doesn't look needed, removed.

File: pkgs/applications/networking/browsers/yandex-browser/default.nix [hunk 6]
```diff
@@ -118,15 +142,20 @@ stdenv.mkDerivation rec {
 
   installPhase = ''
     cp $TMP/ya/{usr/share,opt} $out/ -R
-    substituteInPlace $out/share/applications/yandex-browser-beta.desktop --replace /usr/ $out/
-    ln -sf $out/opt/yandex/browser-beta/yandex_browser $out/bin/yandex-browser
-    ln -sf $out/opt/yandex/browser-beta/yandex_browser $out/bin/yandex-browser-beta
+    cp $out/share/applications/yandex-browser${app}.desktop $out/share/applications/${pname}.desktop || true
+    rm -f $out/share/applications/yandex-browser.desktop
```
Comment (chayleaf) [inline pkgs/applications/networking/browsers/yandex-browser/default.nix:146]: yandex-browser-beta also has a dangling `yandex-browser` symlink in `/opt/yandex/browser-beta`, it isn't exposed to the end user so it's probably fine though

**Review by chayleaf — APPROVED** [review]
LGTM

**Pull request merged**